teps/0152-tekton-conformance-verification.md

```diff
@@ -0,0 +1,219 @@
+
+## Motivation
+
+With the release of [v1 Tekton conformance API spec](https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md), the Tekton community can now offer Tekton conformance verification. The conformance test suite will be the crucial prerequisite for vendors to claim conformance for their service.
+
+### Goals
+- Defines a **detailed conformance verification process** that Tekton vendors can verify and thereby claiming Tekton conformance for their service.
+- Establishes a clear versioning policy for the Conformance Policy, including its verification process, to streamline future updates.
+
+## Proposal
+To establish Tekton conformance offerings for Pipelines from a community perspective, we need to develop a suite of Pipeline YAML test cases and implement a process for verifying the results.
+
+### Conformance Verification Process
+The verification process encompasses both vendor preparation for the conformance certification and community-led verification of conformance results.
+
+#### Pipeline Conformance Offering
+We will develop our own Tekton Pipeline Conformance offering and actively share it within the community once we establish the Task suite for vendor result verification. In order to promote understanding, we need to provide the following:
+1) A clear overview of Conformance ertification benefits.
+2) Step-by-step guidance on the certification process.
+3) A list of certified vendor products.
+
+*Whether the offical release for V1 API is a prerequisite for the intial Tekton Conformance Offering remains to be decided as a community*
+
+The complete conformance verification process for vendors:
+
+1) Prepare:
+
+   Vendors learn about the conformance requirements and the [Conformance Tekton Pipelines API Specification](https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md).
+
+2) Test:
+
+   Vendors integrate their serivce with the Conformance test suite available in Tekton OSS. Run the tests following the instructions provided.
+
+3) Results Submission:
+   
+   Prepare a PR to submit your results to the Conformance repo on GitHub.
+
+4) Tekton Verification:
+   
+   A maintainer from Tekton will review and approve the PR Conformance for verificaiton once all the requirements are met.
+
+#### Vendor Outputs Verification
+
+Vendor output verification will take place within a dedicated 'conformance' repository under the TektonCD organization. We will utilize a combination of GitHub Actions for automation and reviewer judgment to ensure a thorough and precise verification process. Results verification could be automated via a script within a Github Action triggered by vendor output submissions through a Pull Request. The script would analyze the output for keywords indicating test passes, providing vendors with clear feedback on any missing or non-compliant conformance fields. More specifically, the deliverables for vendor test output verification include:
```

Comment:
Why do we need to suggest GHA? We could just use the dogfooding infra that we have.

Reply:
Thanks for the catch, updated. That's what I intended to use but in my mind I felt vendors will see the results on Github.